src/app/api/utils/auth.ts

```diff
@@ -222,7 +226,43 @@ export const authOptions: AuthOptions = {
       }
       if (token.subscriber) {
         session.user.subscriber = token.subscriber;
+
+        // refresh token
+        const dbFxATokens = await getFxATokens(token.subscriber.id);
+        if (
+          !dbFxATokens.fxa_session_expiry ||
+          dbFxATokens.fxa_session_expiry.getTime() < Date.now()
```

Comment:
Maybe we should try rotating it if it expires soon, rather than when it has already expired? Since sessions have a lifetime as well in which the token can expire.

```suggestion
          dbFxATokens.fxa_session_expiry.getTime() < (Date.now() - 5 * 60 * 1000)
```

Reply:
I don't know if we should artificially set a time (not really a best practice)
I think the point is to let sessions expire and potentially force a refresh or relogin

Reply:
I guess the question is: how do we do that? Are we going to refresh the session every time an API request to FxA fails with the relevant status code? Since this callback won't necessarily automatically get called when the session expires.

Reply:
> I guess the question is: how do we do that? Are we going to refresh the session every time an API request to FxA fails with the relevant status code? Since this callback won't necessarily automatically get called when the session expires.

Yes I am wondering that too... I'm not sure what is going to trigger this `session` callback if e.g. a `fetch` to some API (like subscriptions details) fails.